ckan-client-guide/README.md

```diff
@@ -4,113 +4,523 @@ Guide to interacting with [CKAN](/ckan) for power users such as data scientists,
+
+apiKey = '771a05ad-af90-4a70-beea-cbb050059e14'
+apiUrl = 'http://localhost:5000'
+organization = 'datopian'
+dataset = 'dailyprices'
+
+const client = Client(apiKey, organization, dataset, apiUrl)
 ```
 
-### Upload a resource (file) (and implicitly create a new Dataset)
+### Upload a resource
+
+That is to say, upload a file, implicitly creating a new dataset.
+
+#### Python
 
 ```python
-# loads a resource from a path
-resource = f11s.load(resource_file_path)
-dataset_name = 'sample-dataset'
-dataset = f11s.Dataset({'name': dataset_name})
-dataset.add_resource(resource)
-client.push(resource)
+from frictionless import describe
+
+
+resource = describe('my-data.csv')
+client.push_blob(resource)
 ```
 
 ### Create a new empty Dataset with metadata
 
+#### Python
+
 ```python
-dataset = f11s.Dataset({
-    'name': dataset_name,
-    'metadata': {'maintainer_email': 'sample@datopian.com'}
-})
-client.push(dataset)
+client.create('my-data')
+client.push(resource)
 ```
 
 ### Adding a resource to an existing Dataset
 
+::::tip NOTE
+Not implemented yet.
+::::
+
 ```python
-resource = f11s.load('sample.csv')
-
-# this will upload the file and add the resource to the dataset
-# NB: resource metadata e.g. name, title etc will be auto-inferred from the file
-client.push_resource(
-    resource,
-    dataset='dataset-name',
-    append=True
-)
+client.create('my-data')
+client.push_resource(resource)
 ```
 
 ### Edit a Dataset's metadata
 
-> WARNING: Not implemented yet.
+:::tip NOTE
+Not implemented yet.
+:::
 
 ```python
-dataset = client.get_dataset(dataset_name='sample_dataset')
+dataset = client.retrieve('sample-dataset')
 client.update_metadata(
     dataset,
     metadata: {'maintainer_email': 'sample@datopian.com'}
 )
 ```
 
-For details of metadata see the metadata reference below ...
+For details of metadata see the [metadata reference below](#metadata-reference).
 
 ## API - Porcelain
 
-### `retrieve`
+### `Client.create`
+
+Expects as a single argument: a _string_, or a _dict_ (in Python), or an _object_ (in JavaScript). This argument is either a valid dataset name or dictionary with metadata for the dataset in Frictionless format.
+
+### `Client.push`
+
+Expects a single argument: a _dict_ (in Python) or an _object_ (in JavaScript) with a dataset metadata in Frictionless format.
 
-### `push`
+### `Client.retrieve`
+
+Expects a single argument: a string with a dataset name or uniquer ID. Returns a Frictionless resource as a _dict_ (in Python) or as an _Promisse .&lt;object&gt;_ (in JavaScript).
+
+### `Client.push_blob`
+
+Expects a single argument: a _dict_ (in Python) or an _object_ (in JavaScript) with a Frictionless resource.
 
 ## API - Plumbing
 
-### Action API
+### `Client.action`
+
+This method bridges access to the CKAN API _action endpoint_.
+
+#### In Python
+
+Arguments:
+
+| Name                 | Type       | Default    | Description                                                  |
```

Comment:
I would really avoid markdown tables unless auto generated for us. Just used a bullet list for argument info.